flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/ScalarSqlFunction.scala

```diff
@@ -106,7 +106,9 @@ object ScalarSqlFunction {
         val operandTypeInfo = getOperandTypeInfo(callBinding)
 
         val foundSignature = getEvalMethodSignature(scalarFunction, operandTypeInfo)
-          .getOrElse(throw new ValidationException(s"Operand types of could not be inferred."))
+          .getOrElse(throw new ValidationException(
+            s"[${scalarFunction.getClass}]'s Operand types of [${operandTypeInfo}] " +
```

Comment:
I think the message can be improve like this , I copied the code from `createOperandTypeChecker` in this class. 
```
throw new ValidationException(
              s"Given parameters of function '$name' do not match any signature. \n" +
                s"Actual: ${signatureToString(operandTypeInfo)} \n" +
                s"Expected: ${signaturesToString(scalarFunction, "eval")}")`
```
And I think we need to update AggregateSqlFunction as well if needed.